[Security Solution] Remove SavedObjects client UI for CTI and risk scores module

semd opened the pull request:
## Summary

issue: https://github.com/elastic/kibana/issues/155995


1. Remove SO client from UI.
2. Replace **Kibana** dashboard url with **Security** dashboard url in threat intel overview.

### Checklist

Delete any items that are not applicable to this PR.


- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

semd committed:
relocate tags entpoints to own dir

semd committed:
parametrize tag and dashboard apis

semd committed:
tests

semd committed:
use new api fetching CTI dashboards

angorayc committed:
unit tests

angorayc committed:
Merge remote-tracking branch 'upstream/main' into 155995_remove_so_client_ui_cti

angorayc committed:
fix unit tests

angorayc committed:
update i18n keys

angorayc committed:
fix unit tests

angorayc committed:
Merge branch 'main' into 155995_remove_so_client_ui_cti

angorayc commented on x-pack/plugins/security_solution/public/overview/containers/overview_cti_links/index.tsx:
```diff
@@ -4,92 +4,67 @@
-  type: 'tag',
-  search: TAG_REQUEST_BODY_SEARCH,
-  searchFields: ['name'],
+const CTI_TAG_NAME = 'threat intel';
```
Found the tag I have matches this name, so didn't change it. @elastic/security-detections-response  Could you please confirm this is the right tag.

<img width="1503" alt="Screenshot 2023-05-11 at 09 22 39" src="https://github.com/elastic/kibana/assets/6295984/d800a91e-4bbd-4a2c-aa83-b7074b6020c5">

angorayc commented:
Risk score modules tested locally.

elasticmachine commented:
Pinging @elastic/security-threat-hunting (Team:Threat Hunting)

elasticmachine commented:
Pinging @elastic/security-solution (Team: SecuritySolution)

machadoum commented on x-pack/plugins/security_solution/public/overview/containers/overview_cti_links/index.tsx:
```diff
@@ -4,92 +4,67 @@
+  useEffect(() => {
+    fetch(CTI_TAG_NAME);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
Does `fetch` instance changes? If it doesn't we can add it to the dependency array and get rid of `eslint-disable-next-line`.

angorayc committed:
add abort signal

machadoum commented:
I tested Risk Score and Dashboards and couldn't find any problem.

semd committed:
address aborting

angorayc committed:
fix unit tests

semd committed:
fix test

semd committed:
conflict

semd committed:
Merge remote-tracking branch 'upstream/main' into 155995_remove_so_client_ui_cti

semd committed:
some fixes

semd committed:
prevent create tag concurrence

semd commented:
@machadoum thanks for taking the time to review it. We have addressed the abort controls using `useFetch`. 
I retested the Threat Intel dashboard links on the Overview page, and the Security dashboards page. Everything works well.

Regarding race conditions, I created the `DashboardsContext` to prevent exactly that, we were having concurrence when different components were calling the "get or create tag" operation at the same time. 

Now the `useFetchSecurityTags` hook is only called from one place. However, I see the main Dashboard router is always rendered twice when loaded (it comes from up in the tree), causing two consecutive calls. The first call seems to be always aborted, but I am not sure we can rely on aborts only to make sure we never end up with 2 concurrent `createTag` requests. So I've added an additional `tagCreated` reference control:

https://github.com/elastic/kibana/blob/a5ad81273ccb8436bca749fa3b15a77a8cdc1141/x-pack/plugins/security_solution/public/dashboards/containers/use_fetch_security_tags.ts#L34-L36

so now I am pretty sure that it can never happen 👍

angorayc committed:
update unit tests

angorayc committed:
Merge branch '155995_remove_so_client_ui_cti' of github.com:semd/kibana into 155995_remove_so_client_ui_cti
